### PR TITLE
DHCP: Add support to emit and retrieve POP3 server

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -1386,6 +1386,15 @@
             read via <function>sd_network_link_get_sip_servers()</function> function.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>UsePOP3Server=</varname></term>
+          <listitem>
+            <para>When true (the default), the POP3 servers received
+            from the DHCP server will be saved at the state files and can be
+            read via <function>sd_network_link_get_pop3_servers()</function> function.</para>
```
Comment (keszybz) [inline man/systemd.network.xml]: Do we need to make this conditional? I'm pretty sure we won't use that for anything internally, so we can simply always export the data, and the external consumer can use it or not as they see fit.

File: man/systemd.network.xml [hunk 2]
```diff
@@ -1903,6 +1912,19 @@
...
+        <term><varname>EmitPOP3Server=</varname></term>
+        <term><varname>POP3Server=</varname></term>
+
+        <listitem><para>Similar to the <varname>EmitDNS=</varname> and
+        <varname>DNS=</varname> settings described above, these
+        settings configure whether and what POP3 server information
+        shall be emitted as part of the DHCP lease. The same syntax,
+        propagation semantics and defaults apply as for
+        <varname>EmitDNS=</varname> and
+        <varname>DNS=</varname>.</para></listitem>
```
Comment (keszybz) [inline man/systemd.network.xml:1913]: Do we need `EmitPOP3Server=` at all? I'd say that if `POP3Server=` is configured, we should use it unconditionally.

File: src/network/networkctl.c [hunk 3]
```diff
@@ -1749,6 +1751,9 @@ static int link_status_one(
         if (r < 0)
                 return r;
         r = dump_list(table, "NTP:", ntp);
+        if (r < 0)
+                return r;
+        r = dump_list(table, "POP3_SERVERS:", pop3_server);
```
Comment (keszybz) [inline src/network/networkctl.c]: "POP3 servers:"

**Comment by ssahani** [comment]
Updated thanks for the review @keszybz .PTAL

File: src/libsystemd-network/sd-dhcp-lease.c [hunk 4]
```diff
@@ -601,6 +613,12 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const void *option, void
                         log_debug_errno(r, "Failed to parse SIP server, ignoring: %m");
                 break;
 
+        case SD_DHCP_OPTION_POP3_SERVER:
+                r = lease_parse_in_addrs(option, len, &lease->pop3_server, &lease->pop3_server_size);
+                if (r < 0)
+                        log_debug_errno(r, "Failed to parse log server, ignoring: %m");
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp-lease.c]: "pop3 server"

File: src/libsystemd-network/sd-dhcp-lease.c [hunk 5]
```diff
@@ -616,7 +616,7 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const void *option, void
         case SD_DHCP_OPTION_POP3_SERVER:
                 r = lease_parse_in_addrs(option, len, &lease->pop3_server, &lease->pop3_server_size);
                 if (r < 0)
-                        log_debug_errno(r, "Failed to parse log server, ignoring: %m");
+                        log_debug_errno(r, "Failed to parse POP3 server, ignoring: %m");
```
Comment (keszybz) [inline src/libsystemd-network/sd-dhcp-lease.c:619]: oh, ok

File: src/network/networkd-manager.c [hunk 6]
```diff
@@ -1516,6 +1518,10 @@ static int manager_save(Manager *m) {
        if (!sip)
                 return -ENOMEM;
 
+       log = ordered_set_new(&string_hash_ops);
+       if (!log)
+               return -ENOMEM;
+
```
Comment (keszybz) [inline src/network/networkd-manager.c:1524]: Indentation is borked here, and for `sip` above. Why `log` — that's very misleading. Please call is something else.

**Pull request merged**